decent_copy.go

```diff
@@ -16,7 +21,7 @@ func Copy(filepathOrigin, filepathDestiny string) error {
 
 	io.Copy(destFile, srcFile)
 
-	err := destFile.Sync()
+	_ = os.Chmod(filepathDestiny, srcInfo.Mode())
```

Comment:
I think here you can take the error and validate it.. 
What do you think?

Reply:
I agree, but we will either need a test that generates the error, or we need to configure Codecov to not enforce 100% coverage.

I will make the change and then look into whether I can generate/mock the error in tests.

Reply:
I believe that one of these has to happen in order to test this:
Option 1: Mock `os.Chmod`
Option 2: Make `os.Chmod` fail immediately _after_ we have just created `filepathDestiny`

I'm not sure how to do either of these, so I believe we should just pass the error along implicitly (as is done with `io.Copy`). I'll make the change.

Reply:
Looks like a similar issue was encountered with `io.Copy`, and the decision there was to remove the error handling altogether.

In my opinion, it is more correct to return the error in both cases, even if test coverage does not validate the `if err != nil { return err }` part.

Reply:
Ok, I found this library: https://github.com/dave/courtney

It appears to have been created for exactly this situation. Basically, it just excludes simple error return statements from code coverage.

I added it to this PR and it seems to work as intended. Let me know what you think.

Reply:
Hi @hugocarreira, I believe this is ready for a follow up review.